pygyro/advection/__pyccel__/accelerated_advection_steps.f90

```diff
@@ -82,314 +80,296 @@ pure subroutine poloidal_advection_step_expl(f, dt, v, rPts, qPts, nPts, &
-
-
-    do while (endPts_k1_q(j, i) < 0) 
-      endPts_k1_q(j, i) = 2*3.14159265358979d0 + endPts_k1_q(j, i)
+do i = 0_8, nPts(0_8)-1_8, 1_8
+  do j = 0_8, nPts(1_8)-1_8, 1_8
+    !Step one of Heun method
+    !x' = x^n + f(x^n)
+    drPhi_0(j, i) = drPhi_0(j, i) / rPts(j)
+    dthetaPhi_0(j, i) = dthetaPhi_0(j, i) / rPts(j)
+    endPts_k1_q(j, i) = qPts(i) - drPhi_0(j, i) * multFactor
+    endPts_k1_r(j, i) = rPts(j) + dthetaPhi_0(j, i) * multFactor
+    !Handle theta boundary conditions
+    do while (endPts_k1_q(j, i) < 0_8)
+      endPts_k1_q(j, i) = endPts_k1_q(j, i) + 2_8 * 3.14159265358979d0
     end do
-    do while (endPts_k1_q(j, i) > 2*3.14159265358979d0) 
-      endPts_k1_q(j, i) = -2*3.14159265358979d0 + endPts_k1_q(j, i)
-
-
+    do while (endPts_k1_q(j, i) > 2_8 * 3.14159265358979d0)
+      endPts_k1_q(j, i) = endPts_k1_q(j, i) - 2_8 * 3.14159265358979d0
     end do
-    if (.not. (endPts_k1_r(j, i) > rMax .or. endPts_k1_r(j, i) < rPts(0 &
-      ))) then
+    if (.not. (endPts_k1_r(j, i) < rPts(0_8) .or. endPts_k1_r(j, i) > &
+      rMax)) then
       !Add the new value of phi to the derivatives
       !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
       !^^^^^^^^^^^^^^^
       drPhi_k(j, i) = eval_spline_2d_scalar(endPts_k1_q(j, i), &
       endPts_k1_r(j, i), kts1Phi, deg1Phi, kts2Phi, deg2Phi, coeffsPhi, &
-      0, 1)
-      drPhi_k(j, i) = drPhi_k(j, i)/endPts_k1_r(j, i)
-
-
+      0_8, 1_8)
+      drPhi_k(j, i) = drPhi_k(j, i) / endPts_k1_r(j, i)
       dthetaPhi_k(j, i) = eval_spline_2d_scalar(endPts_k1_q(j, i), &
       endPts_k1_r(j, i), kts1Phi, deg1Phi, kts2Phi, deg2Phi, coeffsPhi, &
-      1, 0)
-      dthetaPhi_k(j, i) = dthetaPhi_k(j, i)/endPts_k1_r(j, i)
+      1_8, 0_8)
+      dthetaPhi_k(j, i) = dthetaPhi_k(j, i) / endPts_k1_r(j, i)
     else
       drPhi_k(j, i) = 0.0d0
       dthetaPhi_k(j, i) = 0.0d0
     end if
-    endPts_k2_q(j, i) = modulo(multFactor_half*(-drPhi_0(j, i) - drPhi_k &
-      (j, i)) + qPts(i),2*3.14159265358979d0)
-    endPts_k2_r(j, i) = multFactor_half*(dthetaPhi_0(j, i) + dthetaPhi_k &
-      (j, i)) + rPts(j)
-
-
+    !Step two of Heun method
+    !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
+    endPts_k2_q(j, i) = MODULO((qPts(i) - (drPhi_0(j, i) + drPhi_k(j, i &
+      )) * multFactor_half),(2_8 * 3.14159265358979d0))
+    endPts_k2_r(j, i) = rPts(j) + (dthetaPhi_0(j, i) + dthetaPhi_k(j, i &
+      )) * multFactor_half
   end do
-
+  !Find value at the determined point
+  !Find value at the determined point
+  !Step one of Heun method
+  !x' = x^n + f(x^n)
+  !Handle theta boundary conditions
+  !Add the new value of phi to the derivatives
+  !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
+  !^^^^^^^^^^^^^^^
+  !Step two of Heun method
+  !x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
+  !Clipping is one method of avoiding infinite loops due to
+  !boundary conditions
+  !Using the splines to extrapolate is not sufficient
+  !Find value at the determined point
```

Comment:
It looks like these comments are moved to the wrong place, inside the loop. Do we have an issue for this in Pyccel?

Reply:
I actually hadn't spotted this. The comments come from other functions, it is a coincidence that they are the same comments used in this function. The comments for this function are correctly placed. I have created an issue